tests/test_traits.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+from abc import ABC
+from dataclasses import dataclass
+
+import pytest
+
+from xdsl.ir import OpResult, OpTrait, Operation
+from xdsl.irdl import VarOpResult, VarOperand, irdl_op_definition
+from xdsl.utils.exceptions import VerifyException
+from xdsl.dialects.builtin import i32
+
+
+@dataclass(frozen=True)
+class NoResultsTrait(OpTrait):
+
+    def verify(self, op: Operation):
+        if len(op.results) != 0:
+            raise VerifyException(
+                f"Op has {len(op.results)} results, should have none")
+
+
+@dataclass(frozen=True)
+class NoOperandsTrait(OpTrait):
+
+    def verify(self, op: Operation):
+        if len(op.operands) != 0:
+            raise VerifyException(
+                f"Op has {len(op.operands)} operands, should have none")
+
+
+@dataclass(frozen=True)
+class NOperandsTrait(OpTrait):
+    parameter: int
+
+    def verify(self, op: Operation):
```

Comment:
This feels a bit funky to me, shouldn't this be an `irdl` thingy with `Annotated` if it is a requirement on the arguments?

Reply:
Yes, I see what you mean, I'll change the tests to something that makes more sense then